src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -738,6 +739,22 @@ WhereClauseList(FromExpr *fromExpr)
 
 
 /*
+ * OuterJoinClauseList walks over the FROM expression in the query tree, and builds
+ * a list of outer join clauses from the expression tree.
+ */
+List *
+OuterJoinClauseList(FromExpr *fromExpr)
```

Comment:
I was thinking about using the same walker, just add a case for outer join and return only join clauses inside ```ON(...)``` expression.  

We are only using output of ```OuterJoinClauseList``` to determine whether we are joining on partition column. We do not need all the filtering clauses.